src/link/Elf.zig

```diff
@@ -2221,7 +2228,7 @@ pub fn allocateDeclIndexes(self: *Elf, decl: *Module.Decl) !void {
 pub fn freeDecl(self: *Elf, decl: *Module.Decl) void {
     if (self.llvm_ir_module) |_| return;
 
-    const block_list = self.getDeclBlockList(decl);
+    const block_list = self.getDeclBlockList(decl) orelse return;
```

Comment:
I'm really not sure that here we should return from the function if the Decl is still not typed, as before this commit the function `freeDecl` did not check it.

Is it guaranteed that a Decl has an associated TextBlock only if it has already been typed ?